src/uu/readlink/src/readlink.rs

```diff
@@ -7,80 +7,119 @@
 
 // spell-checker:ignore (ToDO) errno
 
-extern crate getopts;
+extern crate clap;
 
 #[macro_use]
 extern crate uucore;
 
+use clap::{App, Arg};
 use std::fs;
 use std::io::{stdout, Write};
 use std::path::PathBuf;
 use uucore::fs::{canonicalize, CanonicalizeMode};
 
 const NAME: &str = "readlink";
 const VERSION: &str = env!("CARGO_PKG_VERSION");
+const ABOUT: &str = "Print value of a symbolic link or canonical file name";
+const OPT_CANONICALIZE: &str = "canonicalize";
+const OPT_CANONICALIZE_MISSING: &str = "canonicalize-missing";
+const OPT_CANONICALIZE_EXISTING: &str = "canonicalize-existing";
+const OPT_ZERO: &str = "zero";
+const OPT_NO_NEWLINE: &str = "no-newline";
+const OPT_QUIET: &str = "quiet";
+const OPT_SILENT: &str = "silent";
+const OPT_VERBOSE: &str = "verbose";
```

Comment:
I suggest alphabetical order unless there is another underlying logical order.

```suggestion
const OPT_CANONICALIZE: &str = "canonicalize";
const OPT_CANONICALIZE_MISSING: &str = "canonicalize-missing";
const OPT_CANONICALIZE_EXISTING: &str = "canonicalize-existing";
const OPT_NO_NEWLINE: &str = "no-newline";
const OPT_QUIET: &str = "quiet";
const OPT_SILENT: &str = "silent";
const OPT_VERBOSE: &str = "verbose";
const OPT_ZERO: &str = "zero";
```